ml-agents/mlagents/trainers/ppo/trainer.py

```diff
@@ -200,16 +199,17 @@ def create_policy(
             self.seed,
             behavior_spec,
             self.trainer_settings,
-            model_path=self.artifact_path,
-            load=self.load,
             condition_sigma_on_obs=False,  # Faster training for PPO
             create_tf_graph=False,  # We will create the TF graph in the Optimizer
         )
 
         return policy
 
     def add_policy(
-        self, parsed_behavior_id: BehaviorIdentifiers, policy: Policy
+        self,
+        parsed_behavior_id: BehaviorIdentifiers,
+        policy: Policy,
+        register_saver: bool = True,
```

Comment:
Is register_saver ever false?

Reply:
this is for unittest

Reply:
Is is just to avoid extra work during the tests? Can we use a mock/patch there instead, so that we don't need to complicate the interface here?

Reply:
No it actually breaks some tests. For example, `test_trainer_increment_step` in `test_ppo.py`, we create a mock policy and called `trainer.add_policy(behavior_id, policy_mock)`. This will trigger the trainer to register it to Saver and the Saver will try to access its .graph and .sess and do `with self.policy.graph.as_default(): ...` which resulted in error. 
Creating an empty tf.Graph() and attach to the mock doesn't work either. TF will complain "No variables to save".

Reply:
Right, that's why you can use mocks so that the saver doesn't do any registration.

In your example, changing the test_trainer_increment_step to 
```
@mock.patch.object(RLTrainer, "create_saver")
@mock.patch("mlagents.trainers.ppo.trainer.PPOOptimizer")
def test_trainer_increment_step(ppo_optimizer, mock_create_saver):
```
will fix the problem. This will cause RLTrainer.create_saver to return a Mock object, so the calls to register won't have any side effects (unless you configure the mock to have a side effect).

You can also do things like
```
    mock_saver = mock.Mock()
    mock_create_saver.return_value = mock_saver
    ...
    mock_saver.register.assert_called()
    mock_saver.initialize_or_load.assert_called()
```
if you want to sanity-check the logic in the test.

If there's a legitimate case for non-test code to not register, then it's OK to keep the flag. But if *only* tests use it, I think you need to remove it and switch the tests to use Mocks.

Reply:
That solved the problem, thanks. Removed the redundant input argument.